nym-vpn-core/crates/nym-vpn-lib/src/tunnel_state_machine/mod.rs

```diff
@@ -225,9 +225,15 @@ impl TunnelSettings {
         }
         if self.gateway_performance_options != other.gateway_performance_options {
             diff.add(TunnelSettingsDiffFields::GatewayPerformanceOptions);
+            // We care about just the mixnet performance setting changing.
+            if self.gateway_performance_options.mixnet_min_performance
+                != other.gateway_performance_options.mixnet_min_performance
+            {
+                diff.add(TunnelSettingsDiffFields::MixnetPerformanceOptions);
+            }
         }
         if self.mixnet_client_config != other.mixnet_client_config {
-            diff.add(TunnelSettingsDiffFields::MixnetClientConfig);
+            diff.add(TunnelSettingsDiffFields::MixnetPerformanceOptions);
```

Comment:
This is very confusing, why MixnetPerformanceOptions?

Reply:
Because of how the mixnet performance options are spread across two fields.